File: NSEC.Music_Player/Media/LocalPlayerController.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;

namespace NSEC.Music_Player.Media
{
    public class LocalPlayerController : IPlayerController
    {
        public void Completed(CustomMediaPlayer player)
        {
            player.Next();
        }

        public void Load(CustomMediaPlayer player, string filepath)
        {
            string path = System.IO.Path.Combine(Global.MusicPath, $"cache.wav");

            if (File.Exists(path))
                File.Delete(path);

            File.Copy(filepath, path, true);
            player.MediaPlayer.SetDataSource(path);
        }

        public void Prepared(CustomMediaPlayer player)
        {
            player.Play();
        }
    }
}

File: app/src/main/java/com/nejman/nsec/music_player/media/LocalPlayerController.java
package com.nejman.nsec.music_player.media;

import android.media.MediaPlayer;

import com.nejman.nsec.music_player.core.loaders.DataLoader;

import java.io.IOException;

public class LocalPlayerController implements IPlayerController {
    @Override
    public void load(MediaPlayer mediaPlayer, String path) throws IOException {
        mediaPlayer.setDataSource(path);
    }

    @Override
    public void loaded(MediaPlayer mediaPlayer) {
        try {
            DataLoader.playerStateLoader.save();
        } catch (Throwable e) {
            e.printStackTrace();
        }
    }
}
